### PR TITLE
chore(rds): add new versions of MariaDB, MySQL and SQL Server engines

### PR DESCRIPTION
**Pull request by wuhkuh**
Added every change as a separate commit, feel free to rebase as you see fit.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by wuhkuh** [comment]
> err  - PROP @aws-cdk/aws-rds.SqlServerEngineVersion.VER_15_00_4043_23_V1: has been removed [removed:@aws-cdk/aws-rds.SqlServerEngineVersion.VER_15_00_4043_23_V1]
> (...)
> Some packages seem to have undergone breaking API changes. Please avoid.

At best this version could point towards the correct version to preserve API compatibility, but keep in mind there's nothing to preserve—[this version simply doesn't exist](https://support.microsoft.com/en-us/topic/kb4518398-sql-server-2019-build-versions-782ed548-1cd8-b5c3-a566-8b4f9e20293a). Preserving compatibility would also nullify the point of type-safe versioning.

Strongly in favor of using the escape hatch (overriding this warning) and patching this up.

File: packages/@aws-cdk/aws-rds/lib/instance-engine.ts [hunk 1]
```diff
@@ -1350,8 +1358,8 @@ export class SqlServerEngineVersion {
   public static readonly VER_15 = SqlServerEngineVersion.of('15.00', '15.00');
   /** Version "15.00.4043.16.v1". */
   public static readonly VER_15_00_4043_16_V1 = SqlServerEngineVersion.of('15.00.4043.16.v1', '15.00');
-  /** Version "15.00.4043.23.v1". */
-  public static readonly VER_15_00_4043_23_V1 = SqlServerEngineVersion.of('15.00.4043.23.v1', '15.00');
+  /** Version "15.00.4073.23.v1". */
+  public static readonly VER_15_00_4073_23_V1 = SqlServerEngineVersion.of('15.00.4073.23.v1', '15.00');
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/instance-engine.ts:1362]: Please, leave the `VER_15_00_4043_23_V1` unchanged, and instead add the new `VER_15_00_4073_23_V1` constant instead.

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @wuhkuh!

**Comment by skinny85** [comment]
> > err  - PROP @aws-cdk/aws-rds.SqlServerEngineVersion.VER_15_00_4043_23_V1: has been removed [removed:@aws-cdk/aws-rds.SqlServerEngineVersion.VER_15_00_4043_23_V1]
> > (...)
> > Some packages seem to have undergone breaking API changes. Please avoid.
> 
> At best this version could point towards the correct version to preserve API compatibility, but keep in mind there's nothing to preserve—[this version simply doesn't exist](https://support.microsoft.com/en-us/topic/kb4518398-sql-server-2019-build-versions-782ed548-1cd8-b5c3-a566-8b4f9e20293a). Preserving compatibility would also nullify the point of type-safe versioning.
> 
> Strongly in favor of using the escape hatch (overriding this warning) and patching this up.

Instead of removing the `VER_15_00_4043_23_V1` constant, deprecate it.

**Pull request merged**